roles/storage-glusterfs/templates/storage-glusterfs.yml.j2

```diff
@@ -1,14 +1,30 @@
 ---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: kubevirt-heketi-secret
+  namespace: {{ namespace }}
+data:
+  key: {{ heketi_admin_key }}
+{% if external_provisioner %}
+type: gluster.org/glusterfile
+{% else %}
+type: kubernetes.io/glusterfs
+{% endif %}
+---
 kind: StorageClass
 apiVersion: storage.k8s.io/v1beta1
 metadata:
   name: kubevirt
+{% if external_provisioner %}
+provisioner: gluster.org/glusterfile
+{% else %}
 provisioner: kubernetes.io/glusterfs
+{% endif %}
 reclaimPolicy: Delete
 parameters:
-  # Enable this when GlusterFS provisioner supports cloning
-  # smartclone: "true"
   resturl: "http://{{ heketi_url | default(omit) }}"
   restuser: admin
-  secretName: "{{ heketi_secret_name }}"
-  secretNamespace: "{{ heketi_namespace }}"
+  secretName: "kubevirt-heketi-secret"
```

Comment:
@jarrpa Can you please explain, I don't follow there are lines [17](https://github.com/kubevirt/kubevirt-ansible/pull/177/files#diff-8c622beef0965e5d350322336d5a5041R17) and [5](https://github.com/kubevirt/kubevirt-ansible/pull/177/files#diff-ef52ba37e7e1aa81b08eadb145fe0514R5), but this one, 28, is a string, not a variable name to substitute a value, why so?

Reply:
`heketi_secret_name` is now only being used to refer to an extant Secret in OpenShift that holds the heketi admin key. That key is then used to create a new Secret called `kubevirt-heketi-secret`, which will be used by this StorageClass.